compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -503,12 +503,16 @@ getTemplateDocs DocCtx{..} typeMap templateInstanceMap =
     -- (possible if choice instances are defined directly outside the template).
     -- This wouldn't be necessary if we used the type-checked AST.
       where dummyDT = ADTDoc { ad_anchor = Nothing
-                             , ad_name    = Typename $ "External:" <> unTypename n
-                             , ad_descr   = Nothing
+                             , ad_name = dummyName n
+                             , ad_descr = Nothing
                              , ad_args = []
                              , ad_constrs = []
                              , ad_instances = Nothing
                              }
+
+            dummyName (Typename "Archive") = Typename "Archive"
```

Comment:
Can we be more specific here and at least check the module name this is coming from, maybe even the package id?

Reply:
Based on my reading of the code, there's no way to be more specific with here without rewriting the way we extract template data from the module. That's probably worth doing at some point.